src/transformers/models/owlvit/convert_owlvit_to_hf.py

```diff
@@ -0,0 +1,335 @@
+    rename_keys.append(("backbone/clip/logit_scale", "owlvit.logit_scale"))
+
+    # projection heads
+    rename_keys.append(("backbone/clip/text/text_projection/kernel", "owlvit.text_projection.weight"))
+
+    # class and box heads
+    rename_keys.append(("backbone/merged_class_token/scale", "layer_norm.weight"))
+    rename_keys.append(("backbone/merged_class_token/bias", "layer_norm.bias"))
+    rename_keys.append(("class_head/Dense_0/kernel", "class_head.dense0.weight"))
+    rename_keys.append(("class_head/Dense_0/bias", "class_head.dense0.bias"))
+    rename_keys.append(("class_head/logit_shift/kernel", "class_head.logit_shift.weight"))
+    rename_keys.append(("class_head/logit_scale/kernel", "class_head.logit_scale.weight"))
+    rename_keys.append(("class_head/logit_scale/bias", "class_head.logit_scale.bias"))
+    rename_keys.append(("class_head/logit_shift/bias", "class_head.logit_shift.bias"))
+    rename_keys.append(("obj_box_head/Dense_0/kernel", "box_head.dense0.weight"))
+    rename_keys.append(("obj_box_head/Dense_0/bias", "box_head.dense0.bias"))
+    rename_keys.append(("obj_box_head/Dense_1/kernel", "box_head.dense1.weight"))
+    rename_keys.append(("obj_box_head/Dense_1/bias", "box_head.dense1.bias"))
+    rename_keys.append(("obj_box_head/Dense_2/kernel", "box_head.dense2.weight"))
+    rename_keys.append(("obj_box_head/Dense_2/bias", "box_head.dense2.bias"))
+
+    # objectness head (only for v2)
+    if "v2" in model_name:
+        rename_keys.append(("objectness_head/Dense_0/kernel", "objectness_head.dense0.weight"))
+        rename_keys.append(("objectness_head/Dense_0/bias", "objectness_head.dense0.bias"))
+        rename_keys.append(("objectness_head/Dense_1/kernel", "objectness_head.dense1.weight"))
+        rename_keys.append(("objectness_head/Dense_1/bias", "objectness_head.dense1.bias"))
+        rename_keys.append(("objectness_head/Dense_2/kernel", "objectness_head.dense2.weight"))
+        rename_keys.append(("objectness_head/Dense_2/bias", "objectness_head.dense2.bias"))
+
+    # fmt: on
+
+    return rename_keys
+
+
+def rename_and_reshape_key(dct, old, new, config):
+    val = dct.pop(old)
+
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "vision" in new:
+        val = val.reshape(-1, config.vision_config.hidden_size)
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "text" in new:
+        val = val.reshape(-1, config.text_config.hidden_size)
+
+    if "patch_embedding" in new:
+        print("Reshaping patch embedding... for", new)
+        val = val.transpose(3, 2, 0, 1)
+    elif new.endswith("weight") and "position_embedding" not in new and "token_embedding" not in new:
+        val = val.T
+
+    if new.endswith("bias"):
+        val = val.reshape(-1)
+
+    dct[new] = torch.from_numpy(np.array(val))
+
+
+@torch.no_grad()
+def convert_owlvit_checkpoint(model_name, pytorch_dump_folder_path, push_to_hub):
```

Comment:
Same comment about converting the checkpoints, the tests should be optional and outside the conversion in a seperate function if absolutely important.